R-proj/src/Rvolume.cpp

```diff
@@ -0,0 +1,155 @@
+// [[Rcpp::depends(BH)]]
+
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 20012-2018 Vissarion Fisikopoulos
+// Copyright (c) 2018 Apostolos Chalkis
+
+//Contributed and/or modified by Apostolos Chalkis, as part of Google Summer of Code 2018 program.
+
+#include <Rcpp.h>
+#include <RcppEigen.h>
+#include "volume.h"
+
+
+template <class HPolytope>
+class converter{
+public:
+    typedef typename HPolytope::NT NT;
+
+    converter() {}
+
+    HPolytope get(Rcpp::Reference P) {
+        Rcpp::NumericMatrix A = P.field("A");
+        Rcpp::NumericVector b = P.field("b");
+
+        int n = A.cols();
+        int m = A.rows();
+        std::vector<std::vector<NT> > Mat(m+1, std::vector<NT>(n+1));
+
+        Mat[0][0] = m; Mat[0][1] = n+1;
+
+        for (int i = 1; i < m+1; ++i) {
+            Mat[i][0] = b(i-1);
+            for (int j = 1; j < n+1; ++j) {
+                Mat[i][j] = -A(i-1, j-1);
+            }
+        }
+
+        HPolytope HP;
+        HP.init(Mat);
+
+        return HP;
+    }
+
+};
+
+
+// [[Rcpp::plugins(cpp11)]]
+// [[Rcpp::export]]
+double Rvolume (Rcpp::Reference P, Rcpp::NumericMatrix A, unsigned int walk_len, double e,
+                Rcpp::NumericVector InnerBall, bool CG, unsigned int win_len,
+                unsigned int N, double C, double ratio, double frac,
+                bool ball_walk, double delta, bool Vpoly, bool Zono,
```

Comment:
it is redundant to have that many `bool` parameters. One way to remove this is to add a field in the R reference object that represents the polytope to give info about its type (i.e. Hpolytope, Vpolytope etc) and use that in the if statements.

Reply:
the above comment applies to all functions in  R-proj/src

Reply:
we now use int type